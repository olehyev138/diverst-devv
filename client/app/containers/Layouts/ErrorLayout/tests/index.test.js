import React from 'react';
import { shallow } from 'enzyme';
import { unwrap } from '@material-ui/core/test-utils';

import { StyledErrorLayout } from '../index';
const ErrorLayoutNaked = unwrap(StyledErrorLayout);

/**
 * TODO:
 *  - test it accepts & renders a child component
 *  - test it renders the applicationlayout as a child
 */


const props = {
  classes: {},
  component: jest.fn()
};

describe('<ErrorLayout />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const wrapper = shallow(<ErrorLayoutNaked {...props} />);

    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const wrapper = shallow(<ErrorLayoutNaked {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
