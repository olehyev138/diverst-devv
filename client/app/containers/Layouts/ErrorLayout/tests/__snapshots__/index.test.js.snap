// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorLayout /> Should render and match the snapshot 1`] = `
<Memo(ApplicationLayout)
  component={[Function]}
/>
`;
